test/upload-download.test.js

```diff
@@ -173,6 +173,15 @@ describe('storage service', function() {
       });
   });
 
+  it('fails to upload using bad file path', function(done) {
```

Comment:
need 4 tests, I think, one for each circumstance, and I guess the "not bad" paths are already covered by the tests?